finesse/gui/interferometer_monitor.py

```diff
@@ -47,53 +48,61 @@ def __init__(self) -> None:
         """Creates the attributes required to view properties monitored by the EM27."""
         super().__init__("EM27 SOH Monitor")
 
+        self._layout = QGridLayout()
+        self._prop_names: list[str] = []
         self._prop_labels: list[QLabel] = []
         self._val_lineedits: list[QLineEdit] = []
         self._data_table: list[EM27Property] = []
         self._num_props = 0
         self._poll_light = LEDIcon.create_poll_icon()
         self._poll_light._timer.timeout.connect(self.poll_server)  # type: ignore
         self._poll_light._timer.start(2000)
-
-    def _add_widgets(self) -> None:
-        """Creates the widgets to view the EM27 properties."""
-        layout = QGridLayout()
-
-        for i, prop in enumerate(self._data_table):
-            prop_label = QLabel(prop.name)
-            self._prop_labels.append(prop_label)
-            layout.addWidget(prop_label, i, 0)
-
-            val_lineedit = QLineEdit()
-            val_lineedit.setText(f"{prop.value:.6f} {prop.unit}")
-            val_lineedit.setReadOnly(True)
-            val_lineedit.setAlignment(Qt.AlignmentFlag.AlignCenter)
-            self._val_lineedits.append(val_lineedit)
-            layout.addWidget(val_lineedit, i, 1)
-
-        layout.addWidget(QLabel("POLL Server"), i + 1, 0)
-        layout.addWidget(self._poll_light, i + 1, 1)
+        self._layout.addWidget(QLabel("POLL Server"), 0, 0)
+        self._layout.addWidget(self._poll_light, 0, 1)
         self._poll_light.setSizePolicy(
             QSizePolicy.Expanding, QSizePolicy.Fixed  # type: ignore
         )
-
-        self.setLayout(layout)
-
-    def _update_widgets(self) -> None:
-        """Updates the widgets with the latest values."""
-        for i, prop in enumerate(self._data_table):
-            self._prop_labels[i].setText(prop.name)  # in case order changes?
-            self._val_lineedits[i].setText(f"{prop.value:.6f} {prop.unit}")
+        self.setLayout(self._layout)
+
+    def _display_props(self) -> None:
+        """Creates and populates the widgets to view the EM27 properties."""
+        for prop in self._data_table:
+            if prop.name not in self._prop_names:
```

Comment:
This is looking much better! Good job :smile: 

Aside from the minor comments I've made below, I think this function is a little long. I'd put most of it into a new function that returns a `QLineEdit` based on the name you give it.

It could look something like:

```py
if name in self._my_dict:
    return self._my_dict[name]

# ... otherwise create a new control as you're doing now, then return it
```